src/lib/io/include/sol-uart.h

```diff
@@ -102,12 +105,47 @@ struct sol_uart_config {
 #define SOL_UART_CONFIG_API_VERSION (1) /**< compile time API version to be checked during runtime */
     uint16_t api_version; /**< must match #SOL_UART_CONFIG_API_VERSION at runtime */
 #endif
+    /**
+     * @brief Callback containing data was read from UART
+     *
+     * @param data User data as provided in sol_uart_config::user_data
+     * @param uart The UART handle
+     * @param buf The UART data
+     *
+     * @return The number of bytes read from @c buf or negative errno on error.
```

Comment:
Here is user obligation to return numbers of bytes read, isn't it? If so, wouldn't be better to make it clear? On first sight I didn't have this impression - but maybe I'm a bit slow... hehe